src/decoration.c

```diff
@@ -1,27 +1,2687 @@
+    } else
+    {
+        gUnknown_020391A9 = AddPseudoFieldObject(0xc2, SpriteCallbackDummy, 0x88, 0x48, 0);
+    }
+    gSprites[gUnknown_020391A9].oam.priority = 1;
+    DestroySprite(&gSprites[gUnknown_020391A8]);
+    gUnknown_020391A8 = gUnknown_03004880.unk4;
+    gSprites[gUnknown_020391A8].oam.priority = 1;
+}
+
+void sub_8100E70(u8 taskId)
+{
+    s16 *data;
+    data = gTasks[taskId].data;
+    switch (data[2])
+    {
+        case 0:
+            if (!gPaletteFade.active)
+            {
+                sub_80FF0E0(taskId);
+                data[2] = 1;
+                data[6] = 1;
+                data[5] = 1;
+                sub_8072DEC();
+            }
+            break;
+        case 1:
+            SetUpPuttingAwayDecorationPlayerAvatar();
+            pal_fill_black();
+            data[2] = 2;
+            break;
+        case 2:
+            if (sub_807D770() == TRUE)
+            {
+                data[12] = 1;
+                sub_8100EEC(taskId);
+            }
+            break;
+    }
+}
+
+void sub_8100EEC(u8 taskId)
+{
+    MenuZeroFillWindowRect(0, 0, 29, 19);
+    gSprites[gUnknown_020391A8].data7 = 0;
+    gSprites[gUnknown_020391A8].invisible = 0;
+    gSprites[gUnknown_020391A8].callback = sub_8101698;
+    gSprites[gUnknown_020391A9].pos1.x = 0x88;
+    gSprites[gUnknown_020391A9].pos1.y = 0x48;
+    gTasks[taskId].data[10] = 0;
+    gTasks[taskId].func = sub_8100494;
+}
+
+void sub_8100F88(u8 taskId)
+{
+    gTasks[taskId].data[10] = 0;
+    sub_810045C();
+    sub_8101024(taskId);
+}
+
+void sub_8100FB4(u8 taskId)
+{
+    gTasks[taskId].data[10] = 0;
+    sub_810045C();
+    gSprites[gUnknown_020391A8].invisible = 0;
+    gSprites[gUnknown_020391A8].callback = SpriteCallbackDummy;
+    DisplayItemMessageOnField(taskId, gSecretBaseText_StopPuttingAwayDecor, sub_810156C, 0);
+}
+
+void sub_8101024(u8 taskId)
+{
+    u8 mtBehavior;
+    s16 *data;
+    sub_8101460(taskId);
+    if (gUnknown_02039234 != 0)
+    {
+        DisplayItemMessageOnField(taskId, gSecretBaseText_ReturnDecor, sub_8101518, 0);
+    } else
+    {
+        data = gTasks[taskId].data;
+        mtBehavior = MapGridGetMetatileBehaviorAt(data[0], data[1]);
+        if (MetatileBehavior_IsSecretBasePC(mtBehavior) == TRUE || sub_805738C(mtBehavior) == TRUE)
+        {
+            gSprites[gUnknown_020391A8].invisible = 0;
+            gSprites[gUnknown_020391A8].callback = SpriteCallbackDummy;
+            DisplayItemMessageOnField(taskId, gSecretBaseText_StopPuttingAwayDecor, sub_810156C, 0);
+        } else
+        {
+            DisplayItemMessageOnField(taskId, gSecretBaseText_NoDecor, sub_81010F0, 0);
+        }
+    }
+}
+
+void sub_81010F0(u8 taskId)
+{
+    if ((gMain.newKeys & A_BUTTON) || (gMain.newKeys & B_BUTTON))
+    {
+        sub_8100EEC(taskId);
+    }
+}
+
+void sub_8101118(u8 decorIdx, struct UnkStruct_020391B4 *unk_020391B4)
+{
+    if (gDecorations[decorIdx].decor_field_12 == 0)
```

Comment:
As mentioned on Discord. `decor_field_12` seems to indicate the shape, and `var01` and `var02` indicate the width/height.